### PR TITLE
[W-13898974] add written feedback

### PR DESCRIPTION
**Pull request by cheungaryk**
ref: [W-13898974](https://gus.lightning.force.com/a07EE00001Xn40rYAB)

add written feedback portion in the feedback form based on Salesforce design.

Screenshot 2023-08-10 at 11 51 51 AM

### CODE REVIEW

File: src/js/04-feedback.js [hunk 1]
```diff
@@ -4,200 +4,200 @@
   const feedbackCard = document.querySelector('section.feedback-section')
   if (!feedbackCard) return
 
+  const formSubmitAPIVersion = 'v1'
+
+  const questionsMap = {
+    yes: {
+      legend: 'Thanks for the feedback! What made this article helpful? (Please check at least 1 checkbox)',
```
Comment (jdomingu) [inline src/js/04-feedback.js:11]: Tiny nit: Consider adding a period 

`checkbox.)`

File: src/js/04-feedback.js [hunk 2]
```diff
@@ -4,200 +4,200 @@
...
+      is_accurate: 'Contains accurate information',
+      is_comprehensive: 'Includes all of the information I need',
+      is_clear: 'Easy to understand, with clear explanations and visuals',
+      other: 'Something else',
+    },
+    no: {
+      legend: "We're sorry to hear that. How can we improve this article? (Please check at least 1 checkbox),",
```
Comment (jdomingu) [inline src/js/04-feedback.js:18]: Ditto with the period.

File: src/partials/feedback-form/written-feedback-section.hbs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<div class="feedback-form hide">
+    <form>
+        <span aria-live="polite" class="validation-text hide" id="checkboxes-validation-text">Please check at least 1
+            checkbox.</span>
+        <fieldset></fieldset>
+        <p>
+            <label for="comment">Would you like to share any additional feedback? (Optional)</label>
+            <textarea maxlength="1000" placeholder="Tell us what you liked or didn’t like" type="text" id="comment"
```
Comment (jdomingu) [inline src/partials/feedback-form/written-feedback-section.hbs:8]: Tiny nit: Consider adding a period

`didn't like.`

**Review by jdomingu — APPROVED** [review]
Apologies in advance if some of this contradicts the design docs you got. (I can't access the Figma from the work item.) 

That said, most of this is minor visual stuff, so feel free to pass if you don't have time. 

**1. When you click Yes, the thumb icon disappears and the text changes size** 

After clicking yes:

Screenshot 2023-10-18 at 5 15 27 PM

After clicking no:

Screenshot 2023-10-18 at 5 15 21 PM

It would be nice if they were consistent.

**2. There's no hover state for the Submit button.**

It would be nice if this changed color on hover. Related to this: would it be helpful if we used the colors from the existing Salesforce help form?

**3. Textarea elements typically don't have a hover.**

Not sure if this was intentional, but the black border on hover looks a bit odd. The color change on focus looks good though.

Screenshot 2023-10-18 at 5 36 02 PM

**4. You can't change your yes/no selection after clicking**

I found it a bit confusing that after you click yes/no the buttons are still visible, but you can't click them anymore. I think ideally, the buttons would either still be functional or get hidden.

File: src/js/04-feedback.js (hunk 1 above)
Comment (cheungaryk) [reply src/js/04-feedback.js]: added

File: src/partials/feedback-form/written-feedback-section.hbs (hunk 3 above)
Comment (cheungaryk) [reply src/partials/feedback-form/written-feedback-section.hbs]: Added period as requested.

**Comment by jdomingu** [comment]
Thanks for your thorough response! It looks great, and everything you wrote makes sense 👍 

For Figma, signing in with Google did it--I shouldn't have assumed the sign-in-with-SSS button was the way to go 😅

**Review by jdomingu — DISMISSED** [review]
Looks good! Out of curiosity: why are some of the atomic styles in the coveo.css file and others and others in the handlebars template?

**Comment by cheungaryk** [comment]
> Looks good! Out of curiosity: why are some of the atomic styles in the coveo.css file and others and others in the handlebars template?

When I put the atomic styles in coveo.css initially, the styles aren't recognized. I suspect that it is because atomic generates the elements using [shadow root](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot). Definitely annoying and something to revisit later.

**Review by jdomingu — APPROVED** [review]

**Pull request merged**